utils/migration/ckan_django_utils.py

```diff
@@ -0,0 +1,731 @@
+                    query_parameters={
+                        "$entity_Id: ID": entity_id,
+                    },
+                )
+        return id
+
+    def create_license(self):
+        r, id = self.create_update(
+            mutation_class="CreateUpdateLicense",
+            mutation_parameters={
+                "slug": "desconhecida",
+                "name": "desconhecida",
+                "url": "desconhecida.com",
+            },
+            query_class="allLicense",
+            query_parameters={"$slug: String": "desconhecida"},
+        )
+
+        return id
+
+    def create_bq_type(self, obj):
+        r, id = self.create_update(
+            mutation_class="CreateUpdateBigQueryTypes",
+            mutation_parameters={
+                "name": obj,
+            },
+            query_class="allBigquerytypes",
+            query_parameters={"$name: String": obj},
+        )
+
+        return id
+
+    def create_directory_columns(self, column, table_id):
+        if column.get("covered_by_dictionary") == "yes":
+            r, id = self.get_id(
+                query_class="allColumn",
+                query_parameters={
+                    "$name: String": column.get("directory_column").get("column_name"),
+                    "$table_Id: ID": table_id,
+                },
+            )
+            return id
+        else:
+            return None
+
+    def create_temporal_coverage(self, resource, coverage_id):
+        temporal_temporal_coverages = (
+            [None]
+            if resource.get("temporal_coverage") == []
+            or resource.get("temporal_coverage") is None
+            else resource.get("temporal_coverage")
+        )
+
+        if temporal_temporal_coverages != [None]:
+            temporal_temporal_coverages_chain = [
+                s.split(",") if "," in s else [s] for s in temporal_temporal_coverages
+            ]
+            temporal_temporal_coverages = list(
+                itertools.chain(*temporal_temporal_coverages_chain)
+            )
+
+        for temporal_coverage in temporal_temporal_coverages:
+            if temporal_coverage is not None:
+                resource_to_temporal_coverage = parse_temporal_coverage(
+                    temporal_coverage
+                )
+                resource_to_temporal_coverage["coverage"] = coverage_id
+
+                r, id = self.create_update(
+                    query_class="allDatetimerange",
+                    query_parameters={"$coverage_Id: ID": coverage_id},
+                    mutation_class="CreateUpdateDateTimeRange",
+                    mutation_parameters=resource_to_temporal_coverage,
+                )
+
+    def create_coverage(self, resource, coverage):
+        coverage_type = list(coverage.keys())[0]
+        coverage_value = list(coverage.values())[0]
+        area_slugs = (
+            ["desconhecida"]
+            if resource.get("spatial_coverage") == []
```

Comment:
As áreas (incluindo `Desconhecida`) já não são criadas usando o json `bd_spatial_coverage_tree`, usando a chave `unknown`? Não sei se estou confundindo, mas me parece que as áreas já estarão no banco e não precisarão ser recriadas, apenas buscadas e, onde não existir, usar a área `desconhecida`

Reply:
é pq eu tava criando só o que vem nos metadados, ja to escrevendo o codigo pra criar a partir dos avaliable_options